platform/security/policy/security-policy-context/src/main/java/org/codice/ddf/security/policy/context/impl/PolicyManager.java

```diff
@@ -216,30 +246,137 @@ public synchronized void setPolicies(Map<String, Object> properties) {
+        }
+        LOGGER.debug("Policy store initialized, now contains {} entries", policyStore.size());
+    }
 
-            for (Map.Entry<String, List<String>> context : contexts) {
-                List<ContextAttributeMapping> mappings = contextToAttr.get(context.getKey());
-                if (mappings == null) {
-                    mappings = new ArrayList<>();
-                }
-                policyStore.put(context.getKey(),
-                        new Policy(context.getKey(), contextToRealm.get(context.getKey()),
-                                context.getValue(), mappings));
+    private void setPolicyStore(Map<String, String> allContextsToRealms,
+            Map<String, List<String>> allContextsToAuths,
+            Map<String, List<ContextAttributeMapping>> allContextsToAttrs) {
+
+        //add default context values if they do not exist
+        if (allContextsToRealms.get("/") == null) {
+            allContextsToRealms.put("/", DEFAULT_REALM_CONTEXT_VALUE);
+        }
+
+        if (allContextsToAttrs.get("/") == null) {
+            allContextsToAttrs.put("/", new ArrayList<ContextAttributeMapping>());
+        }
+
+        if (allContextsToAuths.get("/") == null) {
+            allContextsToAuths.put("/", new ArrayList<String>());
+        }
+
+        //gather all given context paths
+        Set<String> allContextPaths = new HashSet<>();
+        allContextPaths.addAll(allContextsToRealms.keySet());
+        allContextPaths.addAll(allContextsToAuths.keySet());
+        allContextPaths.addAll(allContextsToAttrs.keySet());
+
+        //resolve all realms, authorization types & required attributes
+        for (String path : allContextPaths) {
+            String contextRealm = getContextRealm(path, allContextsToRealms);
+            List<String> contextAuthTypes = getContextAuthTypes(path, allContextsToAuths);
+            List<ContextAttributeMapping> contextReqAttrs = getContextReqAttrs(path,
+                    allContextsToAttrs);
+            policyStore
+                    .put(path, new Policy(path, contextRealm, contextAuthTypes, contextReqAttrs));
+        }
+    }
+
+    /**
+     * Gets the context realm of the given path. If context realm of given path does not exist it rolls back until a parent path exists with a context realm
+     * If no parent path exists, the context defaults to "/" realm
+     *
+     * @param path           - Path associated to context
+     * @param contextToRealm - Map of all paths and their context realms
+     * @return context - Realm associated to path
+     */
+    public String getContextRealm(String path, Map<String, String> contextToRealm) {
+        String entry = contextToRealm.get(path);
+        if (entry != null) {
+            return entry;
+        } else {
+
+            //continue splitting by last "/"  down path until value found
+            int idx = path.lastIndexOf("/");
+            if (idx <= 0) {
+                idx++;
```

Comment:
Why would I return it here? If I have /A/B/C I need to try to roll back to /A/B and /A before I return the default value for /